### DISCUSSION TITLE
Support for MEOL layer?

### DISCUSSION THREAD

**Discussion by GuzTech** [The-OpenROAD-Project/OpenROAD#1871] [Q&A]
Hello everyone!

I'm in the process of bringing up a new platform which is going well. I am running into a problem with a local interconnect layer which is specified as a `MASTERSLICE` with `LEF58_TYPE` = `MEOL`, so it is not specified as a `ROUTING` layer as is the case with the sky130 PDK. OpenDB complains that there is a parse mismatch for this `lil` layer.

If I add this `lil` layer as a routing layer to `make_tracks.tcl` then OpenDB complains that it is not a routing layer. If I don't add `lil` as a routing layer, then TritonRoute gives an error that it cannot find layer `lil` for viarule that connects `lil` to the first metal layer.

Did anyone else run into a similar problem? If so, how did you solve it?

**Comment by maliberty** [comment]
Please include the odb message.  You should not add tracks if it is not a routing layer.

**Reply by GuzTech** [reply]
If I don't specify the `lil` layer as a routing layer then OpenDB gives me the following warning:

```
[WARNING ODB-0279] parse mismatch in layer property LEF58_TYPE for layer lil : "MEOL"
```

The section in the LEF file looks like this:

```
LAYER lil
	TYPE MASTERSLICE;
	PROPERTY LEF58_TYPE "TYPE MEOL;";
END lil
```

But then TritonRoute gives me an error:

```
[ERROR DRT-0129] Unknown layer lil for viarule lil_m1.
```

The part in the LEF file:

```
VIARULE lil_m1 GENERATE 
	LAYER lil;
		XXX
	LAYER m1;
		XXX
	LAYER ctc;
		XXX
END lil_m1 
```

where the `ctc` layer is of type `CUT`.

**Comment by maliberty** [comment]
We need to add support for MEOL.  You can workaround it my making it a masterslice layer temporarily.

**Reply by maliberty** [reply]
Opened https://github.com/The-OpenROAD-Project/OpenROAD/issues/1872

**Reply by GuzTech** [reply]
Which layer do you mean exactly? `lil` is already a masterslice and `ctc` is a cut layer.
